src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -26,7 +32,34 @@ public DeleteCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_EMPTY_USAGE));
         }
 
-        return StringUtil.isName(args) ? parseByName(args) : parseByIndex(args);
+        if (StringUtil.isName(args)) {
+            return parseByName(args);
+        } else if ((StringUtil.isTag(args))) {
```

Comment:
Not very significant change, I think ((StringUtil.isTag(args))) can be just (StringUtil.isTag(args)), there's an extra bracket.